Resize meta box pane without `ResizableBox`

stokesman opened the pull request:
## What?
Makes meta box pane resizing implementation based on `@use-gesture/react`’s drag hook.

## Why?
- To fix the focus order so it matches the visual order
- For improved dragging feels – this performs better in my testing

Fixes #66436

## How?
Privately exports `useDrag` from the components package and uses it for resizing the meta box pane.

## Testing Instructions
1. In the Post Editor with meta boxes present
2. Resize the meta box pane and ensure it works as before

### Testing Instructions for Keyboard
1. In the Post Editor with meta boxes present
2. Tab through the interface and note that the "Drag to resize" separator/button matches the visual order (i.e. it comes after the editor canvas but before any of the meta box tab stops)
3. With the separator focused use the up and down arrows keys to resize the pane

## Screenshots or screencast <!-- if applicable -->

### Dragging performance

| Before | After |
|--------|--------|
| <video src="https://github.com/user-attachments/assets/bf80ef85-44ab-49ab-829e-66b79c835372"></video>|<video src="https://github.com/user-attachments/assets/56852864-2b57-4863-866c-9259adbe122d"></video>| 

> [!NOTE]
> The recordings above are with Chrome and the browser dev tools open. On trunk, the combination of dev tools being open and the Yoast plugin reliably makes the dragging chunky on my machine. On this branch, with the same conditions it’s much better.

stokesman committed:
Use drag hook from @use-gesture via private api of components package

Mamaduka commented:
Thanks for the "hotfix" PR, @stokesman!

As you noted in the issue, landing a general fix for `ResizableBox` is better for general solutions. Should we focus our efforts on #65637 and follow up with minor fixes?

stokesman committed:
Let drag hook handle keyboard support

stokesman commented:
I'm all for #65637 moving forward. Yet even if it does, I think this is worth considering because:
- this lightly reduces complexity for this use case (or at least allows more consolidated markup/styling).
- the feature added here to fully expand/collapse the pane by "swiping" would be more complex to implement with `ResizableBox` (not that I'm sold on that gesture as being the best way to provide the feature).
- I think there may be a performance advantage too as I’ve noticed chunky resizing in trunk in certain contexts and last time I replaced `ResizableBox` with `useDragging` (briefly in #65466 before reverting those changes) the performance was better. That’s something I’d still need to prove here since a different hook is used.

t-hamano reviewed (commented):
Sorry for the late response to this PR. I'd be happy to move this PR forward towards 7.0. What do you think?

- In WordPress 7.0, the post editor is planned to run as an iframe, so this meta box will become even more prominent.
- Unfortunately, attempts to apply a fix to the upstream re-resizable library itself were unsuccessful. See https://github.com/WordPress/gutenberg/pull/65637#issuecomment-2458919448
- The current meta box implementation appears to be overly customized based on `ResizableBox`.

stokesman committed:
Merge refs to avoid unidiomatic extra element

stokesman committed:
Fix aria-valuenow not updating from toggling pane

stokesman committed:
Revise comment for clarity/accuracy

stokesman committed:
Remove outdated portion of comment

stokesman committed:
Use drag hook’s `filterTaps` option for toggle-if-not-dragged logic

stokesman committed:
Fix initial render height in short viewports

This also fixes a more obscure bug where the preferred height was
not restored after resizing the viewport to less than the “short”
breakpoint and back again.

t-hamano commented on packages/components/src/private-apis.ts:
```diff
@@ -41,6 +46,7 @@ lock( privateApis, {
 	DateRangeCalendar,
 	TZDate,
 	Picker,
+	useDrag,
```
Can we add a changelog that the new private API is exposed?

t-hamano commented on packages/components/src/private-apis.ts:
```diff
@@ -1,3 +1,8 @@
+/**
+ * External dependencies
+ */
+import { useDrag } from '@use-gesture/react';
```
I thought we could use [the `useDragging` hook](https://github.com/WordPress/gutenberg/blob/bceabb971c43e4d066a09013f282f54d41f3e885/packages/compose/src/hooks/use-dragging/index.js#L19), but it doesn't support touch events...

t-hamano commented on packages/edit-post/src/components/layout/style.scss:
```diff
@@ -75,28 +79,16 @@
-			@media not (prefers-reduced-motion) {
-				transition: width 0.3s ease-out;
-			}
+			transition: width 0.3s ease-out;
```
Can we restore `@media not (prefers-reduced-motion)` ?

stokesman replied:
Thanks! I can’t imagine that was intentional and I probably messed up when fixing conflicts.

t-hamano reviewed (approved):
Sorry for the late reply! I think it works perfectly.

stokesman committed:
Restore reduced motion media query

stokesman committed:
Add components package changelog entry

stokesman committed:
Replace function used only in render with variable assignment

stokesman committed:
Fix components package changelog entry
